File: level1/Kaitenzushi/Kaitenzushi.go
package main
import "bufio"
import "fmt"
import "os"
import "strconv"
import "strings"
// Write any import statements here

func contains(elems []int32, v int32) bool {
    for _, s := range elems {
        if v == s {
            return true
        }
    }
    return false
}

func getMaximumEatenDishCount(N int32, D []int32, K int32) int32 {
  // Write your code here
	var result int32 = 0
	var i int32 = 0
	prev := make([]int32, K)
	for ; i < N; i++ {
		if contains(prev, D[i]) == false {
			prev = prev[1:]
			prev = append(prev, D[i])
			result++
		}
	}
	return result
}

File: level1/Kaitenzushi/Kaitenzushi.php
<?php

function getMaximumEatenDishCount(int $N, array $D, int $K): int {
    // Write your code here
    $numDishes = 0;
    $lastAteDishAt = [];
    foreach ($D as $dish) {
        if (isset($lastAteDishAt[$dish]) && $K >= $numDishes - $lastAteDishAt[$dish]) {
            continue;
        }

        $lastAteDishAt[$dish] = $numDishes;
        $numDishes++;
    }
    return $numDishes;
}

//function getMaximumEatenDishCount(int $N, array $D, int $K): int {
//    // Write your code here
//    $prevDishes = array_fill(0, $K, 0);
//    $result = 0;
//    foreach ($D as $d) {
//        if (!in_array($d, $prevDishes)) {
//            $result++;
//        }
//        array_shift($prevDishes);
//        array_push($prevDishes, $d);
//    }
//    return $result;
//}
